src/cmd/tools/read_data_files/main/main.go

```diff
@@ -184,53 +186,60 @@ func main() {
 
 			startTime := xtime.ToUnixNano(time.Now())
 			endTime := xtime.FromSeconds(0)
-			numDatapointsPerSeries := 0
+			datapointsPerSeries := 0
 			sumPerSeries := 0.0
-			if benchMode != benchmarkSeries {
-				iter := m3tsz.NewReaderIterator(xio.NewBytesReader64(data), true, encodingOpts)
+			var timestamps, values strings.Builder
+			timestamps.WriteByte('t')
+			values.WriteByte('v')
+
+			seriesCount++
+			iter := m3tsz.NewReaderIterator(xio.NewBytesReader64(data), true, encodingOpts)
+			if benchMode != benchmarkNone {
 				fmt.Printf("%d\t%s\t", shard, id.String())
-				for iter.Next() {
-					dp, _, annotation := iter.Current()
+			}
+			for iter.Next() {
+				datapointCount++
+
+				dp, _, _ := iter.Current()
+				datapointsPerSeries++
+				if benchMode == benchmarkSeries {
+					sumPerSeries += dp.Value
 					startTime = xtime.MinUnixNano(startTime, dp.TimestampNanos)
-                    endTime = xtime.MaxUnixNano(endTime, dp.TimestampNanos)
-                    numDatapointsPerSeries++
-                    sumPerSeries += dp.Value
-					annotationSizeTotal += uint64(len(annotation))
-					datapointCount++
-					if dp.Value == 0 {
-						zeroDatapointCount++
-					}
+					endTime = xtime.MaxUnixNano(endTime, dp.TimestampNanos)
+				} else if benchMode == benchmarkDatapoints {
+					timestamps.WriteString(fmt.Sprintf(",%d", dp.TimestampNanos.ToTime().UTC().Second()))
```

Comment:
Not important. `TimestampNanos` is a unix timestamp at nanosecond resolution. Just need `dp.TimestampNanos / 1,000,000,000`.

https://pkg.go.dev/github.com/m3db/m3/src/dbnode/ts#Datapoint.TimestampNanos

Reply:
yep, good catch, this could avoid method chains and improve performance.